src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1,21 +1,40 @@
 /// For all the linkers we support, and information they might
 /// need out of the shared crate context before we get rid of it.
```

Comment:
cc @alexcrichton @michaelwoerister I feel like this module would be better suited for another crate, say `rustc_codegen_link` or `rustc_link`(`er`), what do you think?

Reply:
Who should depend on who? `CrateInfo` and `CodegenResults` are both needed by `rustc_codegen_ssa` and the linker code.